deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -219,6 +219,11 @@ public Response create(
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("This stage does not allow deploying a private build. Please Contact #teletraan and #security-related to whitelist your stage for deploying private build"));
         }
         
+        // check if the build is from a trusted artifact url
+        if(!buildBean.getArtifact_url().startsWith("https://soxrepo.pinadmin.com") && envBean.getEnsure_trusted_build()) {
+            throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("This stage does not allow deploying a build from untrusted url. Please Contact #teletraan and #security-related to whitelist your sox url for deploying"));
```

Comment:
please change "Contact ..... " to ensure the build artifact is published to a trusted url."

Reply:
Updated :)